core/backend/src/test/standalone/TxnManager.test.ts

```diff
@@ -920,3 +945,116 @@ describe("TxnManager", () => {
+      assert.strictEqual(txn.hasUnsavedChanges, hasUnsavedChanges, `hasUnsavedChanges: ${message}`);
+      assert.strictEqual(txn.hasPendingTxns, hasPendingTxns, `hasPendingTxns: ${message}`);
+      assert.strictEqual(txn.isUndoPossible, isUndoPossible, `isUndoPossible: ${message}`);
+      assert.strictEqual(txn.isRedoPossible, isRedoPossible, `isRedoPossible: ${message}`);
+    }
+
+    it("Pull with pending Txns", async () => {
+      for (const txn of txns) {
+        assertTxnStates(txn, false, false, false, false, false, "initial state");
+      }
+      // This test checks various states of local briefcases during pull
+      // [0]: source imodel which pushes out the changes
+      // [1]: briefcase which has no changes
+      // [2]: briefcase with saved changes
+      // [3]: briefcase with unsaved changes (then save, undo and pull)
+
+      IModelTestUtils.createAndInsertPhysicalPartitionAndModel(briefcases[0], IModelTestUtils.getUniqueModelCode(briefcases[0], "firstPhysicalModel"), false);
+
+      assertTxnStates(txns[0], true, true, false, false, false, "initial create");
+      briefcases[0].saveChanges("inserted first model");
+      assertTxnStates(txns[0], true, false, true, true, false, "initial saved changes");
+
+      await briefcases[0].pushChanges({ accessToken: accessTokens[0], description: "new physical model" });
+      assertTxnStates(txns[0], false, false, false, false, false, "pushed changes");
+
+      // [1] model has no changes
+      await briefcases[1].pullChanges();
+      assertTxnStates(txns[1], false, false, false, false, false, "pulled changes");
+
+      // [2] model has saved changes
+      IModelTestUtils.createAndInsertPhysicalPartitionAndModel(briefcases[2], IModelTestUtils.getUniqueModelCode(briefcases[2], "localPhysicalModel"), false);
+      briefcases[2].saveChanges("inserted local model");
+      assertTxnStates(txns[2], true, false, true, true, false, "briefcase 2 with saved local changes");
+      assert.strictEqual(txns[2].getUndoString(), "inserted local model");
+
+      await briefcases[2].pullChanges();
+      if(pullMergeMethod === "Merge") {
+        assert.strictEqual(txns[2].getUndoString(), "Merged");
+        assertTxnStates(txns[2], true, false, true, true, false, "briefcase 2 after pull");
+      } else {
+        assertTxnStates(txns[2], true, false, true, false, false, "briefcase 2 after pull");
+      }
+      assert.strictEqual(2, getCount(briefcases[2], PhysicalModel.classFullName)); // the pulled and local models
+
+      // [3] model gets unsaved changes
+      IModelTestUtils.createAndInsertPhysicalPartitionAndModel(briefcases[3], IModelTestUtils.getUniqueModelCode(briefcases[3], "localPhysicalModel"), false);
+      assertTxnStates(txns[3], true, true, false, false, false, "briefcase 3 with unsaved local changes");
+      await assertThrowsAsync(async () => briefcases[3].pullChanges(), "cannot processed due to unsaved changes.");
```

Comment:
should be "cannot be processed due to unsaved changes"
I did not find the string both in itwin and imodel-native....